### PR TITLE
swagger changes for containerservice event resources

### PR DESCRIPTION
**Pull request by Supriya-R-Dixit**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/eventgrid/data-plane/Microsoft.ResourceNotifications/stable/2018-01-01/ContainerServiceEventResources.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-01-01",
+    "title": "Schema of Container Service events published to Azure Event Grid via Azure Resource Notifications",
+    "description": "Describes the schema of the Container Service events published to Azure Event Grid. This corresponds to the Data property of an EventGridEvent."
+  },
+  "paths": {},
+  "definitions": {
+    "ResourceNotificationsContainerServiceEventResourcesScheduledEventEmittedEventData": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.ResourceNotifications/stable/2018-01-01/ContainerServiceEventResources.json:10]: This name is very long. Can we try something like ResourceNotificationsContainerServiceResourceScheduledEventData?

Comment (aydany) [reply specification/eventgrid/data-plane/Microsoft.ResourceNotifications/stable/2018-01-01/ContainerServiceEventResources.json:10]: It is a really long event, but we cannot shorten it without taking the risk for a conflict in the future. These names are constructed as follows:
- ResourceNotifications - prefix
- TopicType: ContainerServiceEventResources
- EventType: ScheduledEventEmitted
- EventData: i believe this is EG's convention

If we shorten any of TopicType and EventType, we will run the risk of having a conflict, e.g., we expose ContainerServiceResources topic with the Scheduled eventType.

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.ResourceNotifications/stable/2018-01-01/ContainerServiceEventResources.json:10]: I guess the EventType being ScheduledEventEmitted is confusing to me. Can you give more details as to what this event represents? From the name, it is very hard to infer what it is supposed to represent.

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.ResourceNotifications/stable/2018-01-01/ContainerServiceEventResources.json:10]: > If we shorten any of TopicType and EventType, we will run the risk of having a conflict, e.g., we expose ContainerServiceResources topic with the Scheduled eventType.

How about `ResourceNotificationsContainerServiceEventResourcesScheduledEventData`? This maintains the full topic type and simply shortens the event type.

Comment (aydany) [reply specification/eventgrid/data-plane/Microsoft.ResourceNotifications/stable/2018-01-01/ContainerServiceEventResources.json:10]: The event type is called "ScheduledEventEmitted". 

"Scheduled Events" are a customer facing feature. Those used to be available only from inside VMs and with this work customers can now subscribe to them via system topic: https://learn.microsoft.com/en-us/azure/virtual-machines/windows/scheduled-events.

"Scheduled" is not a replacement for "ScheduledEvent". 

This was called ScheduledEventEmitted to follow the guidance from Event Grid team of having a "thing" and an "action on the thing": the thing is ScheduledEvent and the action is Emitted.

Comment (aydany) [reply specification/eventgrid/data-plane/Microsoft.ResourceNotifications/stable/2018-01-01/ContainerServiceEventResources.json:10]: As discussed over teams we will go with: ```ResourceNotificationsContainerServiceEventResourcesScheduledEventData```

**Comment by qiaozha** [comment]
@JoshLove-msft  You may check here to get yourself unblocked https://github.com/Azure/azure-rest-api-specs/pull/31271/checks?check_run_id=32680563961 The JS generation failure is not blocking this pr. and it's a known issue https://github.com/Azure/autorest.typescript/issues/2513 for us can be easily fixed by manually remove those types while preparing the JS SDK release PR.

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**